src/main/java/org/opensearch/sdk/handlers/ExtensionActionRequestHandler.java

```diff
@@ -83,15 +83,17 @@ public RemoteExtensionActionResponse handleRemoteExtensionActionRequest(Extensio
         logger.debug("Found matching action [" + action.name() + "], an instance of [" + action.getClass().getName() + "]");
 
         // Extract request class name from bytes and instantiate request
-        int nullPos = indexOf(request.getRequestBytes(), RemoteExtensionActionRequest.UNIT_SEPARATOR);
-        String requestClassName = new String(Arrays.copyOfRange(request.getRequestBytes(), 0, nullPos + 1), StandardCharsets.UTF_8)
-            .stripTrailing();
+        int nullPos = indexOf(request.getRequestBytes().toByteArray(), RemoteExtensionActionRequest.UNIT_SEPARATOR);
```

Comment:
Instead of converting ByteString to ByteArray multiple times, can we do it once and use the newly created ByteArray everywhere?

Reply:
There's an open PR on OpenSearch to make this a util class... eventually may want to update here.

Reply:
Makes sense. Done